examples/slack-security-findings/daily_findings.py

```diff
@@ -0,0 +1,188 @@
+from __future__ import annotations
+
+import json
+import os
+import sys
+from datetime import date, timedelta
+from http.client import RemoteDisconnected
+from json import JSONDecodeError
+from string import Template
+from typing import TypedDict, Callable, Any
+from urllib import request
+from urllib.error import URLError
+import logging
+from argparse import ArgumentParser
+
+LOG = logging.getLogger(__name__)
+
+
+class Finding(TypedDict):
+    href: str
+    first_seen_snapshot_date: date
+    file_path: str
+    start_line: int
+    end_line: int
+    type: str
+    severity: str
+    severity_score: float
+    status: str
+
+
+class SigridApiClient:
+
+    def __init__(self, customer: str, system: str, token: str):
+        sigrid_api_endpoint = 'https://sigrid-says.com/rest/analysis-results/api/v1/security-findings/$customer/$system'
+        self.sigrid_api = Template(sigrid_api_endpoint).substitute(customer=customer, system=system)
```

Comment:
Wouldn't it be better to use an f-string here? AFAIK they now the preferred way of doing template strings in Python.

Reply:
Much better, changed it!